core/include/seeding/track_params_estimation_helper.hpp

```diff
@@ -78,7 +90,7 @@ inline TRACCC_HOST_DEVICE bound_vector seed_to_bound_vector(
     scalar B = uv2[1] - A * uv2[0];
 
     // Curvature (with a sign) estimate
-    scalar rho = -2.0 * B / std::hypot(1., A);
+    scalar rho = -2.0 * B / math_ns::hypot(float(1.), A);
```

Comment:
As mentioned above, we can consider `algebra::math::perp`

Reply:
Std::hypot and std::perp are not the same things though. We would have to teach algebra-plugins about std::hypot first.

Reply:
They don't make a same output?

Reply:
:facepalm: I was mistaking `algebra::getter::perp` for one of the functions I added recently in:

https://github.com/acts-project/algebra-plugins/blob/main/math/common/include/algebra/math/common.hpp

Since I did check that none of those was `std::hypot`.

You're right, the `perp(...)` function from algebra-plugins does calculate this. But that function always works on the `vector2` (or `vector3`) type of whatever storage we are using. Still... maybe you're right. Maybe a more elegant thing would be to replace this line with:

```c++
scalar rho = -2.0 * B / algebra::getter::perp({1., A});
```

:thinking: Could you give that a try Konrad?

Reply:
Unfortunately it doesn't work out of the box: 
```shell
In file included from /home/kkusiak/traccc-test/traccc/core/include/seeding/track_params_estimation.hpp:10,
                 from /home/kkusiak/traccc-test/traccc/examples/run/cpu/seq_example.cpp:18:
/home/kkusiak/traccc-test/traccc/core/include/seeding/track_params_estimation_helper.hpp: In function ‘traccc::bound_vector traccc::seed_to_bound_vector(const spacepoint_container_t&, const traccc::seed&, const vector3&, traccc::scalar)’:
/home/kkusiak/traccc-test/traccc/core/include/seeding/track_params_estimation_helper.hpp:93:58: error: no matching function for call to ‘perp(<brace-enclosed initializer list>)’
     scalar rho = -2.0 * B / algebra::getter::perp({1., A});
                                                          ^
In file included from /home/kkusiak/traccc-test/build_nvidia/_deps/algebraplugins-src/math/cmath/include/algebra/math/impl/cmath_cylindrical2.hpp:11,
                 from /home/kkusiak/traccc-test/build_nvidia/_deps/algebraplugins-src/math/cmath/include/algebra/math/cmath.hpp:12,
                 from /home/kkusiak/traccc-test/build_nvidia/_deps/algebraplugins-src/frontend/array_cmath/include/algebra/array_cmath.hpp:11,
                 from /home/kkusiak/traccc-test/traccc/plugins/algebra/include/traccc/array/include/plugins/algebra/array_definitions.hpp:13,
                 from /home/kkusiak/traccc-test/traccc/core/include/definitions/primitives.hpp:11,
                 from /home/kkusiak/traccc-test/traccc/io/include/io/csv.hpp:19,
                 from /home/kkusiak/traccc-test/traccc/examples/run/cpu/seq_example.cpp:11:
/home/kkusiak/traccc-test/build_nvidia/_deps/algebraplugins-src/math/cmath/include/algebra/math/impl/cmath_getter.hpp:51:37: note: candidate: ‘template<class size_type, template<class, size_type <anonymous> > class array_t, class scalar_t, size_type N, typename std::enable_if<(N >= 2), bool>::type <anonymous> > scalar_t algebra::cmath::perp(const array_t<scalar_t, N>&)’
 ALGEBRA_HOST_DEVICE inline scalar_t perp(
                                     ^~~~
/home/kkusiak/traccc-test/build_nvidia/_deps/algebraplugins-src/math/cmath/include/algebra/math/impl/cmath_getter.hpp:51:37: note:   template argument deduction/substitution failed:
In file included from /home/kkusiak/traccc-test/traccc/core/include/seeding/track_params_estimation.hpp:10,
                 from /home/kkusiak/traccc-test/traccc/examples/run/cpu/seq_example.cpp:18:
/home/kkusiak/traccc-test/traccc/core/include/seeding/track_params_estimation_helper.hpp:93:58: note:   couldn't deduce template parameter ‘size_type’
     scalar rho = -2.0 * B / algebra::getter::perp({1., A});
```

Reply:
Lookes like `brace-enclosed initializer list` is not deduced. Have you tried sth like `vector2{1.,A}` instead?

Reply:
That indeed works :+1: